packages/data-api/src/__tests__/testFetchEvents.js

```diff
@@ -35,48 +34,54 @@ export const testFetchEvents = () => {
   const api = new TupaiaDataApi(getTestDatabase());
 
   const assertCorrectResponse = async (options, responses) =>
-    expect(api.fetchEvents(options)).to.eventually.deep.equal(
+    expect(api.fetchEvents(options)).resolves.toStrictEqual(
       getEventsFromResponses(responses, options.dataElementCodes),
     );
 
   it('throws an error with invalid parameters', async () => {
-    await expect(api.fetchEvents()).to.be.rejectedWith(/provide options/);
-    await expect(api.fetchEvents(null)).to.be.rejectedWith(/provide options/);
-    await expect(api.fetchEvents({})).to.be.rejectedWith(/Invalid content/);
-
-    // no surveyCode
-    await expect(
-      api.fetchEvents({
-        organisationUnitCodes: ['NZ_AK', 'NZ_WG'],
-        dataElementCodes: ['BCD1', 'BCD325'],
-      }),
-    ).to.be.rejectedWith(/Invalid content.*surveyCode/);
-
-    // no organisationUnitCodes
-    await expect(
-      api.fetchEvents({
-        surveyCode: 'BCD',
-        dataElementCodes: ['BCD1', 'BCD325'],
-      }),
-    ).to.be.rejectedWith(/Invalid content.*organisationUnitCodes/);
-
-    // no dataElementCodes
-    await expect(
-      api.fetchEvents({
-        surveyCode: 'BCD',
-        organisationUnitCodes: ['NZ_AK', 'NZ_WG'],
-      }),
-    ).to.be.rejectedWith(/Invalid content.*dataElementCodes/);
-
-    // invalid startDate format
-    await expect(
-      api.fetchEvents({
-        surveyCode: 'BCD',
-        organisationUnitCodes: ['NZ_AK', 'NZ_WG'],
-        dataElementCodes: ['BCD1', 'BCD325'],
-        startDate: 'January first, 2020',
-      }),
-    ).to.be.rejectedWith(/Invalid content.*startDate/);
+    const testData = [
+      [undefined, /provide options/],
+      [null, /provide options/],
+      [{}, /Invalid content/],
+      [
+        {
+          organisationUnitCodes: ['NZ_AK', 'NZ_WG'],
+          dataElementCodes: ['BCD1', 'BCD325'],
+        },
+        /Invalid content.*surveyCode/,
+      ], // no surveyCode
+      [
+        {
+          surveyCode: 'BCD',
+          dataElementCodes: ['BCD1', 'BCD325'],
+        },
+        /Invalid content.*organisationUnitCodes/,
+      ], // no organisationUnitCodes
+      [
+        {
+          surveyCode: 'BCD',
+          organisationUnitCodes: ['NZ_AK', 'NZ_WG'],
+        },
+        /Invalid content.*dataElementCodes/,
+      ], // no dataElementCodes
+      [
+        {
+          surveyCode: 'BCD',
+          organisationUnitCodes: ['NZ_AK', 'NZ_WG'],
+          dataElementCodes: ['BCD1', 'BCD325'],
+          startDate: 'January first, 2020',
+        },
+        /Invalid content.*startDate/,
+      ], // invalid startDate format
+    ];
+
+    for (const [options, expectedRegexp] of testData) {
+      try {
+        await api.fetchEvents(options);
+      } catch (e) {
+        expect(e.message).toEqual(expect.stringMatching(expectedRegexp));
+      }
+    }
```

Comment:
Since the issue hasn't resolved ([beyondessential/tupaia-backlog#1201](https://github.com/beyondessential/tupaia-backlog/issues/1201)). `.ToBeRejectedWith` is not working in this test case.

Reply:
> @billli0 could you update this PR to use `.toBeRejectedWith` since it seems to be fine now? You will have to merge the latest `dev` in this branch

Thanks for pointing it out Kostas! `.toBeRejectedWith` works perfectly now!